#!/usr/bin/env python

PACKAGE="realsense_camera"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#             Name                                         Type      Level  Description                                   Default    Min     Max
gen.add("enable_depth",                                    bool_t,   0,     "Enable Depth",                               True)
gen.add("color_backlight_compensation",                    int_t,    0,     "Backlight Compensation",                     0,         0,      4)
gen.add("color_brightness",                                int_t,    0,     "Brightness",                                 0,         0,      255)
gen.add("color_contrast",                                  int_t,    0,     "Contrast",                                   50,        16,     64)
gen.add("color_exposure",                                  int_t,    0,     "Exposure",                                   166,       50,     666)
gen.add("color_gain",                                      int_t,    0,     "Gain",                                       64,        0,      256)
gen.add("color_gamma",                                     int_t,    0,     "Gamma",                                      300,       100,    280)
gen.add("color_hue",                                       int_t,    0,     "Hue",                                        0,         -2200,  2200)
gen.add("color_saturation",                                int_t,    0,     "Saturation",                                 64,        0,      255)
gen.add("color_sharpness",                                 int_t,    0,     "Sharpness",                                  50,        0,      7)

# Must be set only if color_enable_auto_white_balance is disabled
gen.add("color_white_balance",                             int_t,    0,     "White Balance",                              4600,      2000,   8000)

gen.add("color_enable_auto_exposure",                      int_t,    0,     "Enable Auto Exposure",                       1,         0,      1)
gen.add("color_enable_auto_white_balance",                 int_t,    0,     "Enable Auto White Balance",                  1,         0,      1)
gen.add("r200_lr_auto_exposure_enabled",                   int_t,    0,     "LR Auto Exposure Enabled",                   0,         0,      1)
gen.add("r200_lr_gain",                                    int_t,    0,     "LR Gain",                                    400,       100,    6399)

# Must be set only if r200_lr_auto_exposure_enabled is disabled
gen.add("r200_lr_exposure",                                int_t,    0,     "LR Exposure",                                164,       1,      164)

gen.add("r200_emitter_enabled",                            int_t,    0,     "Emitter Enabled",                            0,         0,      1)
gen.add("r200_depth_clamp_min",                            int_t,    0,     "Depth Clamp Min",                            0,         0,      65535)
gen.add("r200_depth_clamp_max",                            int_t,    0,     "Depth Clamp Max",                            65535,     0,      65535)
gen.add("r200_depth_control_estimate_median_decrement",    int_t,    0,     "Depth Control Estimate Median Decrement",    5,         0,      255)
gen.add("r200_depth_control_estimate_median_increment",    int_t,    0,     "Depth Control Estimate Median Increment",    5,         0,      255)
gen.add("r200_depth_control_median_threshold",             int_t,    0,     "Depth Control Median Threshold",             192,       0,      1023)
gen.add("r200_depth_control_score_minimum_threshold",      int_t,    0,     "Depth Control Score Min Threshold",          1,         0,      1023)
gen.add("r200_depth_control_score_maximum_threshold",      int_t,    0,     "Depth Control Score Max Threshold",          512,       0,      1023)
gen.add("r200_depth_control_texture_count_threshold",      int_t,    0,     "Depth Control Texture Count Threshold",      6,         0,      31)
gen.add("r200_depth_control_texture_difference_threshold", int_t,    0,     "Depth Control Texture Difference Threshold", 24,        0,      1023)
gen.add("r200_depth_control_second_peak_threshold",        int_t,    0,     "Depth Control Second Peak Threshold",        27,        0,      1023)
gen.add("r200_depth_control_neighbor_threshold",           int_t,    0,     "Depth Control Neighbor Threshold",           7,         0,      1023)
gen.add("r200_depth_control_lr_threshold",                 int_t,    0,     "Depth Control LR Threshold",                 24,        0,      2047)
gen.add("fisheye_exposure",                                int_t,    0,     "Fisheye Exposure",                           40,        40,     331)
gen.add("fisheye_gain",                                    int_t,    0,     "Fisheye Gain",                               0,         0,      2047)
gen.add("fisheye_strobe",                                  int_t,    0,     "Fisheye Strobe",                             0,         0,      1)
gen.add("fisheye_external_trigger",                        int_t,    0,     "Fisheye External Trigger",                   0,         0,      1)
gen.add("fisheye_enable_auto_exposure",                    int_t,    0,     "Fisheye Enable Auto Exposure",               1,         0,      1)
gen.add("fisheye_auto_exposure_mode",                      int_t,    0,     "Fisheye Auto Exposure Mode",                 0,         0,      2)
gen.add("fisheye_auto_exposure_antiflicker_rate",          int_t,    0,     "Fisheye Auto Exposure Antiflicker Rate",     60,        50,     60)
gen.add("fisheye_auto_exposure_pixel_sample_rate",         int_t,    0,     "Fisheye Auto Exposure Pixel Sample Rate",    1,         1,      3)
gen.add("fisheye_auto_exposure_skip_frames",               int_t,    0,     "Fisheye Auto Exposure Skip Frames",          2,         0,      3)
gen.add("frames_queue_size",                               int_t,    0,     "Frames Queue Size",                          20,        1,      20)
gen.add("hardware_logger_enabled",                         int_t,    0,     "Hardware Logger Enabled",                    0,         0,      1)

exit(gen.generate(PACKAGE, "realsense_camera", "zr300_params"))
